### PR TITLE
Setup git submodules so googletest is fetched, unbreaking alpine in releases

### LINKED ISSUE #4639: GitHub Releases missing Linux binaries
For versions 106 and 107, Mac and Windows binaries are attached to the GitHub Release, but the Linux binary that was present in earlier releases (e.g. https://github.com/WebAssembly/binaryen/releases/download/version_105/binaryen-version_105-x86_64-linux.tar.gz ) is missing.

### PR DESCRIPTION
**Pull request by kripken**
Without this the CMake step fails on not finding

`googletest/googletest/src/gtest_main.cc`

Fixes #4639 

This won't be tested in this PR. What's the best way to test it? Make a fake release I guess?

### CODE REVIEW

File: .github/workflows/create_release.yml [hunk 1]
```diff
@@ -125,6 +125,11 @@ jobs:
     - name: install python dev dependencies
       run: ./alpine.sh pip3 install -r requirements-dev.txt
 
+    - name: git submodules
+      run: |
+        ./alpine.sh git submodule init
+        ./alpine.sh git submodule update
+
```
Comment (sbc100) [inline .github/workflows/create_release.yml:132]: Better to modify the `checkout` step above. e.g:

```
    - uses: actions/checkout@v1                                                     
      with:                                                                         
        submodules: true      
```

**Review by sbc100 — APPROVED** [review]

**Comment by kripken** [comment]
Any idea how to test this? I created a draft release "Testing alpine git integration" but it doesn't seem to start any actions,

https://github.com/WebAssembly/binaryen/releases/tag/untagged-dc99bed2246bfe1042b4 (details only shown in the edit display, https://github.com/WebAssembly/binaryen/releases/edit/untagged-dc99bed2246bfe1042b4)

There seems to be a previous draft release "test_tag" which *does* have builds for it, though. Maybe it's just slow?

**Comment by sbc100** [comment]
That action is triggered by tag creation..  and I don't currently see a tag for your test release (I though releases even drafts were based on tags.. but I guess not).

I'm not sure this change needs testing though.. its fairly obviously correct.

**Comment by kripken** [comment]
Hmm, I did create a tag when I drafted the release. Can see it on https://github.com/WebAssembly/binaryen/releases/edit/untagged-dc99bed2246bfe1042b4 - it is "alpgit". So that is odd...

I'd prefer to test it just to be sure we don't forget this next release, if somehow it doesn't work - the silent missing files from releases (happened in the last 2) are annoying. But yeah, this seems trivial, so if I can't figure it out I'll leave it I guess.

**Comment by sbc100** [comment]
I looks like the draft release process creates the tag on publish.. so there is no tag as of yet.

**Comment by kripken** [comment]
Oh, ok... I guess I'll publish it then, then delete it.

**Comment by kripken** [comment]
Thanks, yes, I see actions are starting now...

**Pull request merged**